File: CCLib/src/java/lang/native/Number.lua
natives["java.lang.Number"] = natives["java.lang.Number"] or { }
natives["java.lang.Number"]["toString(D)Ljava/lang/String;"] = function(value)
    return toJString(tostring(value))
end

natives["java.lang.Number"]["toString(J)Ljava/lang/String;"] = function(value)
    return toJString(tostring(value))
end

natives["java.lang.Number"]["toString(I)Ljava/lang/String;"] = function(value)
    return toJString(tostring(value))
end

natives["java.lang.Number"]["toString(F)Ljava/lang/String;"] = function(value)
    return toJString(tostring(value))
end

File: CCLib/src/java/lang/Number.java
package java.lang;

public abstract class Number {
    static native String toString(long value);
    static native String toString(double value);
    static native String toString(int value);
    static native String toString(float value);
    
    public abstract byte byteValue();
    public abstract short shortValue();
    public abstract int intValue();
    public abstract long longValue();
    public abstract float floatValue();
    public abstract double doubleValue();
}